### PR TITLE
fix: broken search bar with tags filters #727

### LINKED ISSUE #727: Bug: tag filter search feature no longer works on the envelop plugins page with the new new fuzzy matching search bar
Introduced from #674

### PR DESCRIPTION
**Pull request by YassinEldeeb**
closes #727

### CODE REVIEW

**Comment by YassinEldeeb** [comment]
@B2o5T can you please review this? 👀

File: packages/components/src/components/marketplace-search.tsx [hunk 1]
```diff
@@ -36,17 +36,28 @@ export const MarketplaceSearch = ({
...
-          // Mapping the queryList items into a list of strings including the titles
-          queryList.items.map(e => e.title),
-        )
-        .map(e => e.original);
+      // Filter by tags
+      if (query.startsWith('#')) {
+        const filteredItems = queryList.items.filter(item =>
+          query
+            .split(' ')
+            .map(e => e.trim().replace('#', ''))
```
Comment (B2o5T) [inline packages/components/src/components/marketplace-search.tsx:44]:
```suggestion
            .split(/\s+/)
            .map(e => e.replace('#', ''))
```

**Comment by B2o5T** [comment]
@YassinEldeeb I simplified your code to looks more readable https://github.com/the-guild-org/docs/pull/728/commits/30889b76213859879adc3f4aaa2258a63d88421c

let's me know what do you think :)

**Review by B2o5T — APPROVED** [review]

**Comment by YassinEldeeb** [comment]
yep, that looks better. thank you!

**Pull request merged**